modules/javafx.graphics/src/main/java/javafx/scene/Node.java

```diff
@@ -1404,6 +1407,93 @@ public String getName() {
         return visible;
     }
 
+    /**
+     * Indicates whether or not this {@code Node} is shown. A node is considered shown if it's
+     * part of a {@code Scene} that is part of a {@code Window} whose
+     * {@link Window#showingProperty showing property} is {@code true}. The {@link Node#visibleProperty visibility}
+     * of the node or its scene does not affect this property.
+     * <p>
+     * This property can be used in conjunction with {@link ObservableValue#when} to
+     * create bindings which are only actively listening to their source when the node is shown.
+     * <p>
+     * This property can also be used to start animations when the node is shown, and to stop them
+     * when it is no longer shown.
+     *
+     * @since 20
+     */
+    private ReadOnlyBooleanProperty shown;
+
+    public final boolean isShown() {
+        if (shown == null) {  // avoid eager instantiation of property
+            Scene s = getScene();
+            if (s == null) return false;
+            Window w = s.getWindow();
+            return w != null && w.isShowing();
+        }
+
+        return shown.get();
+    }
+
+    public final ReadOnlyBooleanProperty shownProperty() {
```

Comment:
This property probably should have been named `showing` to align it with `Window.showing`, but the "good name" is already taken by `ComboBox` and other controls. Just out of interest, have you considered alternatives to `shown`?

Reply:
I didn't look too far for alternative names after I discovered `showing` would not be possible.  The name comes from `isTreeShowing` which is used for a similar purpose (inside `Node`) and from `conditionOnShowing` in ReactFX.

The name needs to imply that visibility has no effect on it (ie, `setVisible(false)` won't toggle it).  Neither does it check if the node isn't covered or off screen.

In theory you could use a more general name (like `active` as in "part of an active currently showing scene graph").  `isActive` seems to even be available...

A name like "used" or "inUse" may also work (as in "indicates the node is currently used as part of a currently showing scene graph".

Reply:
That reminds me... `parent` is described as:

>  The parent of this {@code Node}. If this {@code Node} has not been added to a scene graph, then parent will be null.

Which I think is incorrect; parent can easily be non-null while not being part of a scene graph.

Reply:
`displayed`?